swagger_spec_validator/validator12.py

```diff
@@ -45,13 +49,23 @@ def validate_spec_url(url):
     """
 
     log.info('Validating %s' % url)
-    resource_listing = json.load(urllib2.urlopen(url, timeout=1))
+    resource_listing = json.load(urllib2.urlopen(url, timeout=TIMEOUT_SEC))
     validate_resource_listing(resource_listing)
 
+    is_file_uri = urlparse(url).scheme == 'file'
+
+    def resource_file_name(resource):
+        assert resource.startswith('/')
+        return resource[1:] + '.json'
+
     for api in resource_listing['apis']:
-        path = url + api['path']
+        if is_file_uri:
+            parent_dir = os.path.dirname(url)
+            path = os.path.join(parent_dir, resource_file_name(api['path']))
+        else:
+            path = url + api['path']
```

Comment:
Could we make this (and the line above assigning to `is_file_uri`) a separate function?

It could even get an inline function like `resource_file_name()`

```python

def get_path(url, api_path):
    ...

```

That way the main function stays really small and easy to read